### PR TITLE
Milestone 3: validations and model specs

### PR DESCRIPTION
**Pull request by Fsher07**
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  it 'photo should be string' do
+    @user.photo = 1
+    expect(@user).not_to be_valid
+  end
+
+  it 'posts_counter should be greater than or equal to 0' do
+    @user.postscounter = -1
+    expect(@user).not_to be_valid
+  end
```
Comment (medaminedev66) [inline spec/models/user_spec.rb:30]: - [ ]  Any custom methods in the User model should have a corresponding unit test such as the `recent_posts` method.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @post.likes_counter = -13
+    expect(@post).to_not be_valid
+
+    @post.likes_counter = 0
+    expect(@post).to be_valid
+
+    @post.likes_counter = 19
+    expect(@post).to be_valid
+  end
+end
```
Comment (medaminedev66) [inline spec/models/post_spec.rb:63]: - [ ]  Following the project requirements, any custom methods in the Post, Comment, and Like models should have a corresponding unit test.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @Fsher07  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- `User` and `Post` models have validations
- There are no linter errors
- You have a descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by RolandM99 — APPROVED** [review]
<h1>Status: Approve 🥇 ✔</h1>
Hi @Fsher07, 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**